### ISSUE TITLE
kubectl get node : no resources found.

### ISSUE THREAD

**Issue by logici** [ubuntu/microk8s#1932]
ENV:
host : mac

- multipass ubuntu:20.04  (one: microk8s-vm-2)
- microk8s 1.19/stable

```
ubuntu@microk8s-vm-2:~$ microk8s kubectl get node
No resources found
ubuntu@microk8s-vm-2:~$ microk8s kubectl get all --all-namespaces
NAMESPACE   NAME                 TYPE        CLUSTER-IP     EXTERNAL-IP   PORT(S)   AGE
default     service/kubernetes   ClusterIP   10.152.183.1   <none>        443/TCP   14h
```

```
ubuntu@microk8s-vm-2:~/inspection-report/snap.microk8s.daemon-apiserver$ cat systemctl.log 
● snap.microk8s.daemon-apiserver.service - Service for snap application microk8s.daemon-apiserver
     Loaded: loaded (/etc/systemd/system/snap.microk8s.daemon-apiserver.service; enabled; vendor preset: enabled)
     Active: active (running) since Fri 2021-01-22 09:09:19 CST; 33min ago
   Main PID: 1253 (kube-apiserver)
      Tasks: 13 (limit: 4683)
     Memory: 509.9M
     CGroup: /system.slice/snap.microk8s.daemon-apiserver.service
             └─1253 /snap/microk8s/1856/kube-apiserver --cert-dir=/var/snap/microk8s/1856/certs --service-cluster-ip-range=10.152.183.0/24 --authorization-mode=AlwaysAllow --service-account-key-file=/var/snap/microk8s/1856/certs/serviceaccount.key --client-ca-file=/var/snap/microk8s/1856/certs/ca.crt --tls-cert-file=/var/snap/microk8s/1856/certs/server.crt --tls-private-key-file=/var/snap/microk8s/1856/certs/server.key --kubelet-client-certificate=/var/snap/microk8s/1856/certs/server.crt --kubelet-client-key=/var/snap/microk8s/1856/certs/server.key --secure-port=16443 --token-auth-file=/var/snap/microk8s/1856/credentials/known_tokens.csv --insecure-port=0 --storage-backend=dqlite --storage-dir=/var/snap/microk8s/1856/var/kubernetes/backend/ --allow-privileged=true --requestheader-client-ca-file=/var/snap/microk8s/1856/certs/front-proxy-ca.crt --requestheader-allowed-names=front-proxy-client --requestheader-extra-headers-prefix=X-Remote-Extra- --requestheader-group-headers=X-Remote-Group --requestheader-username-headers=X-Remote-User --proxy-client-cert-file=/var/snap/microk8s/1856/certs/front-proxy-client.crt --proxy-client-key-file=/var/snap/microk8s/1856/certs/front-proxy-client.key

Jan 22 09:42:35 microk8s-vm-2 microk8s.daemon-apiserver[1253]: E0122 09:42:35.451494    1253 authentication.go:53] Unable to authenticate the request due to an error: invalid bearer token
Jan 22 09:42:36 microk8s-vm-2 microk8s.daemon-apiserver[1253]: E0122 09:42:36.451627    1253 authentication.go:53] Unable to authenticate the request due to an error: invalid bearer token
Jan 22 09:42:37 microk8s-vm-2 microk8s.daemon-apiserver[1253]: E0122 09:42:37.451392    1253 authentication.go:53] Unable to authenticate the request due to an error: invalid bearer token
Jan 22 09:42:38 microk8s-vm-2 microk8s.daemon-apiserver[1253]: E0122 09:42:38.555927    1253 authentication.go:53] Unable to authenticate the request due to an error: invalid bearer token
Jan 22 09:42:39 microk8s-vm-2 microk8s.daemon-apiserver[1253]: E0122 09:42:39.451446    1253 authentication.go:53] Unable to authenticate the request due to an error: invalid bearer token
Jan 22 09:42:40 microk8s-vm-2 microk8s.daemon-apiserver[1253]: E0122 09:42:40.452409    1253 authentication.go:53] Unable to authenticate the request due to an error: invalid bearer token
Jan 22 09:42:41 microk8s-vm-2 microk8s.daemon-apiserver[1253]: E0122 09:42:41.451638    1253 authentication.go:53] Unable to authenticate the request due to an error: invalid bearer token
Jan 22 09:42:41 microk8s-vm-2 microk8s.daemon-apiserver[1253]: E0122 09:42:41.840513    1253 authentication.go:53] Unable to authenticate the request due to an error: invalid bearer token
Jan 22 09:42:41 microk8s-vm-2 microk8s.daemon-apiserver[1253]: E0122 09:42:41.872311    1253 authentication.go:53] Unable to authenticate the request due to an error: invalid bearer token
Jan 22 09:42:42 microk8s-vm-2 microk8s.daemon-apiserver[1253]: E0122 09:42:42.452043    1253 authentication.go:53] Unable to authenticate the request due to an error: invalid bearer token
```

In the know_tokens.csv, there are only four rows not five. It should include `$token,system:node:microk8s-vm-2,kubelet-0,"system:nodes"`.

```
ubuntu@microk8s-vm-2:/var/snap/microk8s/1856/credentials$ cat known_tokens.csv
czBNemZ0d1NsWEFVNFpRQU9XYWc1Ym9FUW9Xd3NIRStya0RudFQyaGVncz0K,system:kube-proxy,kube-proxy
UEF4eU96NnZxOGVoOHo3eFhnSFpIMWtwNFZEUG1tV3FrbGFwVFBraWFJOD0K,system:kube-controller-manager,controller
dEFnOVlTc1hXdUxjWlYybi9LTmtxSW5Zd1JPckx3dktkeStzWHVMRGtQaz0K,system:kube-scheduler,scheduler
VEJ0ZlB1SU12d1dDL3Rsb0hla0NsVFBreVNJSVFXWWM2NjZZOTA3YWlhcz0K,admin,admin,"system:masters"
```

[no-resource-found-inspection-report-20210122_094304.tar.gz](https://github.com/ubuntu/microk8s/files/5863766/no-resource-found-inspection-report-20210122_094304.tar.gz)

At present, I recovered it by reinstall microk8s only.

**Comment by ktsakalozos** [comment]
Hi @logici , thank you for reporting this. On the kubelet logs you attached I see this error:
```
Jan 22 09:42:43 microk8s-vm-2 microk8s.daemon-kubelet[1328]: E0122 09:42:43.867383    1328 kubelet.go:2183] node "microk8s-vm-2" not found
```

I am not sure why the API server was failing to start. Maybe something changed in the storage of the VM? What can you tell us about your setup? Did anything change recently?

**Comment by logici** [comment]
Hi @ktsakalozos, I am sorry that I did not describe the whole process in detail. It should be a cluster including three nodes, `microk8s-vm`, `microk8s-vm-1`, `microk8s-vm-2`. I wanted to test that what would happen when my node crashed. So I stop the `vm-2` and its status in the cluster was `Notready`. And the I removed the node `microk8s-vm-2` in other node. When I started the node `vm-2`, I saw that this node `vm-2` had the cluster info from `microk8s status`, `microk8s kubectl get node`, and `cat cluster.yaml`. So, I executed `microk8s leave` in the `vm-2`. And then the cluster was fine, but the `vm-2` left cluster and its status was what I showed before.
